src/sentry/integrations/msteams/card_builder/__init__.py

```diff
@@ -137,128 +149,105 @@ def build_group_footer(group: Group, rules: Sequence[Rule], event: Event) -> Col
     )
 
 
-def build_group_actions(group, event, rules, integration):
+def build_input_choice_card(
+    data: Any,
+    title: str,
+    input_id: str,
+    choices: Sequence[Tuple[str, Any]],
+    default_choice: Any = None,
+) -> AdaptiveCard:
+    return MSTeamsMessageBuilder().build(
+        title=create_text_block(title, weight=TextWeight.BOLDER),
+        text=create_input_choice_set_block(
+            id=input_id, choices=choices, default_choice=default_choice
+        ),
+        actions=[create_action_block(ActionType.SUBMIT, title=title, data=data)],
+    )
+
+
+def create_issue_action_block(
+    event: Event,
+    rules: Sequence[Rule],
+    integration: Integration,
+    toggled: bool,
+    action: ACTION_TYPE,
+    action_title: str,
+    reverse_action: ACTION_TYPE,
+    reverse_action_title: str,
+    **card_kwargs: Any,
+) -> Action:
+    if toggled:
+        data = generate_action_payload(reverse_action, event, rules, integration)
```

Comment:
What is `reverse_action` in this context? Like undoing something?

Reply:
Yes, like action -> reverse_action -- ignore -> stop_ignoring, assingn -> unassign or resolve -> unresolve. This is the best name I could think of 😄. @ceorourke any suggestions for better names?

Reply:
Maybe `undo_action`? It's not bad as is, I just wanted to clarify.

Reply:
Oh I guess I'll keep `reverse_action` and add some comments.